src/sentry/incidents/endpoints/serializers/workflow_engine_detector.py

```diff
@@ -1,34 +1,339 @@
+                            rpc_component,
+                            None,
+                            action.get("settings"),
+                        )
+                        if rpc_component
+                        else None
+                    )
+                    if component is None:
+                        errors.append({"detail": f"Could not fetch details from {rpc_app.name}"})
+                        action["disabled"] = True
+                        continue
+
+                    action["formFields"] = component.app_schema.get("settings", {})
+
+            if errors:
+                result[detector]["errors"] = errors
+            alert_rule_triggers.append(serialized)
+
+    def add_projects(
+        self, result: DefaultDict[Detector, dict[str, Any]], detectors: dict[int, Detector]
+    ) -> None:
+        detector_projects = set()
+        for detector in detectors.values():
+            detector_projects.add((detector.id, detector.project.slug))
+
+        for detector_id, project_slug in detector_projects:
+            rule_result = result[detectors[detector_id]].setdefault(
+                "projects", []
+            )  # keyerror I guess could be here
+            rule_result.append(project_slug)
+
+    def add_created_by(
+        self, result: DefaultDict[Detector, dict[str, Any]], detectors: Sequence[Detector]
+    ) -> None:
+        user_by_user_id: MutableMapping[int, RpcUser] = {
+            user.id: user
+            for user in user_service.get_many_by_id(
+                ids=[
+                    detector.created_by_id
+                    for detector in detectors
+                    if detector.created_by_id is not None
+                ]
+            )
+        }
+        for detector in detectors:
+            # this is based on who created or updated it during dual write
+            rpc_user = None
+            if detector.created_by_id:
+                rpc_user = user_by_user_id.get(detector.created_by_id)
+            if not rpc_user:
+                result[detector]["created_by"] = {}
+            else:
+                created_by = dict(
+                    id=rpc_user.id, name=rpc_user.get_display_name(), email=rpc_user.email
+                )
+                result[detector]["created_by"] = created_by
+
+    def add_owner(
+        self, result: DefaultDict[Detector, dict[str, Any]], detectors: Sequence[Detector]
+    ) -> None:
+        for detector in detectors:
+            if detector.owner_user_id or detector.owner_team_id:
+                actor = detector.owner
+                if actor:
+                    result[detector]["owner"] = actor.identifier
+
+    def add_latest_incident(
+        self,
+        result: DefaultDict[Detector, dict[str, Any]],
+        user: User | RpcUser | AnonymousUser,
+        detectors: dict[int, Detector],
+        detector_to_action_ids: defaultdict[Detector, list[int]],
+    ) -> None:
+        for detector in detectors.values():
```

Comment:
🤔 it looks like this might have an N+1 query inside of this loop -- could we flatten the lookup here?